ow_plexil/src/plexil-adapter/OwInterface.cpp

```diff
@@ -908,6 +908,41 @@ void OwInterface::taskDiscardSampleAction (int frame, bool relative,
      default_action_done_cb<TaskDiscardSampleResultConstPtr> (Name_TaskDiscardSample));
 }
 
+bool OwInterface::injectSimulatedFault (const std::string &fault_name, double probability) const
+{
+  srand(time(0));
+  double roll = (double) rand()/RAND_MAX;
+  string combined_fault_name = "faults/" + fault_name;
+  if (roll <= probability){
+    if(m_genericNodeHandle->hasParam(combined_fault_name)){
```

Comment:
I think these two checks should be reversed, so that the function returns false if the fault name is invalid, regardless of the roll.  The roll should only determine if the fault is set.  

This comment also applies to the next function.